app/controllers/customers_controller.rb

```diff
@@ -0,0 +1,53 @@
+class CustomersController < ApplicationController
+  before_action :create_customer, only: %i[show edit update destroy]
+
+  def index
+    @customers = Customer.all
+  end
+
+  def show
+    @customer = Customer.find_by_id(params[:id])
```

Comment:
Don't needs to add this: @customer = Customer.find_by_id(params[:id])
As you are already finding customer in before action method.